### PR TITLE
Add fromfile

### PR DESCRIPTION
**Pull request by jakirkham**
For now this does the simple thing of calling `numpy.fromfile` and converting the result to a CuPy array, which is returned. In the future this could be replaced by something more optimal (like https://github.com/cupy/cupy/issues/2576 ;).

### CODE REVIEW

**Comment by leofang** [comment]
LGTM. Just one nitpick: Could you add a doctoring to `fromfile()` and document this fallback behavior? This could serve as a reminder for #2576. 

FYI, there are some efforts on the fallback mode (see the meta issue #2066). Is it possible to take advantage of that feature, if it could be a one-liner like the current implementation? (I never used it myself and I am unaware of its current status, so this is more a question than a suggestion.)

**Comment by jakirkham** [comment]
Of course. Have added a docstring. Please take a look.

I'm not really familiar with that either. So not really sure. Maybe someone else can comment?

File: cupy/creation/from_data.py [hunk 1]
```diff
@@ -137,7 +139,16 @@ def copy(a, order='K'):
...
 
 
-# TODO(okuta): Implement fromfile
+def fromfile(file, dtype=float, count=-1, sep='', offset=0):
+    """Reads an array from a file.
+
+    .. note::
+        Uses NumPy's ``fromfile`` and coerces the result to a CuPy array.
+
+    .. seealso:: func:`numpy.fromfile`
```
Comment (leofang) [inline cupy/creation/from_data.py]:
```suggestion
    .. seealso:: :func:`numpy.fromfile`
```

Comment (jakirkham) [reply cupy/creation/from_data.py]: Thanks!

File: cupy/creation/from_data.py [hunk 2]
```diff
@@ -137,7 +139,16 @@ def copy(a, order='K'):
 # TODO(okuta): Implement frombuffer
 
 
-# TODO(okuta): Implement fromfile
+def fromfile(file, dtype=float, count=-1, sep='', offset=0):
```
Comment (kmaehashi) [inline cupy/creation/from_data.py]: `offset` is new in NumPy 1.17.0, so users running older NumPy cannot use this option.
One idea may be to make the entire argument `*args, **kwargs` and just pass it to `numpy.fromfile`.

Comment (jakirkham) [reply cupy/creation/from_data.py]: SGTM

Comment (leofang) [reply cupy/creation/from_data.py]: Good catch!

**Comment by jakirkham** [comment]
Thanks for the feedback. Have addressed in the latest commit. Please let me know if there is anything else. 🙂

**Comment by emcastillo** [comment]
Would it be possible to add some tests?

**Comment by jakirkham** [comment]
Yeah that seems fair. How do you normally test IO functions here? Is there an example you can point me to?

**Comment by emcastillo** [comment]
Usually the way we do it is to create a tmp file in the setup and delete it in the teardown. But this seems more chainer than cupy.

I guess that you can write some random array to the file in the setup, and then read it back and compare both. But its more testing the parameters and corner cases than the functionality itself as this has been tested in numpy already.

If its too complicated don't bother :).

**Comment by jakirkham** [comment]
Sure have added such a test. Please let me know what you think.

FWIW I tried with an in-memory file at first, but [NumPy doesn't seem to like that]( https://github.com/numpy/numpy/issues/2230#issuecomment-554156223 ).

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by jakirkham** [comment]
Thanks for the reviews! 😄

**Comment by mnicely** [comment]
I think this slipped through documentation. I can't find it [here](https://docs-cupy.chainer.org/en/stable/index.html)

**Comment by jakirkham** [comment]
Please see PR ( https://github.com/cupy/cupy/pull/2626 ).

**Comment by jakirkham** [comment]
This is now in the docs

https://docs-cupy.chainer.org/en/latest/reference/generated/cupy.fromfile.html

**Pull request merged**